indexing-service/src/test/java/org/apache/druid/indexing/overlord/ForkingTaskRunnerTest.java

```diff
@@ -353,4 +353,77 @@ int waitForTaskProcessToComplete(Task task, ProcessHolder processHolder, File lo
     Assert.assertEquals(TaskState.FAILED, status.getStatusCode());
     Assert.assertEquals("task failure test", status.getErrorMsg());
   }
+
+  @Test
+  public void testJavaOptsAndJavaOptsArrayOverride() throws ExecutionException, InterruptedException
+  {
+    ObjectMapper mapper = new DefaultObjectMapper();
+    String javaOpts = "-Xmx1g -Xms1g";
+    List<String> javaOptsArray = ImmutableList.of("-Xmx10g", "-Xms10g");
+    Task task = NoopTask.create().withJavaOptsContext(javaOpts, javaOptsArray);
```

Comment:
suggestion(nit) : we can create this via `mapper.readValue(jsonString, NoopTask.class)` with jsonString containing the contructor params + context values as strings and string-arrays. that would emulate an user input more closely

Reply:
Done